app/src/main/java/de/westnordost/streetcomplete/quests/wheelchair_access/AddWheelchairAccessPublicTransportForm.java

```diff
@@ -21,6 +20,27 @@
 
 	private void setTitle()
 	{
-		setTitle(R.string.quest_wheelchairAccess_public_transport_title);
+		String highway = null;
+		OsmElement element = getOsmElement();
+		if (highway == null) {highway = element != null && element.getTags() != null ? element.getTags().get("highway") : null;}
+		if (highway == null) {highway = element != null && element.getTags() != null ? element.getTags().get("railway") : null;}
+		if (highway == null) {highway = element != null && element.getTags() != null ? element.getTags().get("public_transport") : null;}
+
+		switch (highway) {
+			case "bus_stop":
+				setTitle(R.string.quest_wheelchairAccess_public_transport_bus_stop_title);
+				break;
+			case "platform":
+				setTitle(R.string.quest_wheelchairAccess_public_transport_platform_title);
+				break;
+			case "station":
+				setTitle(R.string.quest_wheelchairAccess_public_transport_station_title);
+				break;
+			case "subway_entrance":
+				setTitle(R.string.quest_wheelchairAccess_public_transport_entrance_title);
+				break;
+			default:
+				setTitle(R.string.quest_wheelchairAccess_public_transport_default_title);
+		}
```

Comment:
Hmm right, makes sense for unnamed platforms. I wish we had a dictionary that spits out the name of any location given a set of tags.

Anyway, if the railway station, bus stop, etc. *has* a name ("name" tag), it should be used. I propose to have this:

```
<string name="quest_wheelchairAccess_title">Is %s accessible by wheelchair?</string>
<string name="element_subway_entrance">subway entrance</string>
<string name="element_railway_station">railway station</string>
<string name="element_platform">platform</string>
<!-- etc. -->
```
And then simply put the element name into the %s. If the element has a name ("name" tag), use that one instead (and put it into quotation marks).